src/store/books/details/Types.tsx

```diff
@@ -0,0 +1,15 @@
+/* eslint-disable no-unused-vars */
+export type Book = {
+  id?: number;
+  title: string;
+  author: string;
+  cover: string;
+};
+
+export interface IBookDetailsContext {
+  book: Book | null;
+  bookId: number | null;
+  loading: boolean;
+  error: string | null;
+  setBookId?: (bookIdParam: number) => void;
```

Comment:
Try setting type as follows: 

`setBookId?: Function;
`
In that case, you don't need `/* eslint-disable no-unused-vars */`